### PR TITLE
HM-232 Caching Danh Mục tổng Hợp

### PR DESCRIPTION
**Pull request by huynhlap13**

### CODE REVIEW

File: app/Console/Kernel.php [hunk 1]
```diff
@@ -14,7 +14,8 @@ class Kernel extends ConsoleKernel
      */
     protected $commands = [
         Commands\PushDmtvt::class,
-        Commands\PushTvtTheoKho::class
+        Commands\PushTvtTheoKho::class,
+        Commands\PushDmth::class
```
Comment (ngkimlong) [inline app/Console/Kernel.php]: Commands\CacheDmth::class

File: app/Services/DanhMucTongHopService.php [hunk 2]
```diff
@@ -6,15 +6,45 @@
 use App\Http\Resources\DanhMucTongHopResource;
 use App\Http\Resources\HanhChinhResource;
 use App\Repositories\DanhMuc\DanhMucTongHopRepository;
+use App\Repositories\Redis\DanhMuc\DmTongHopRedisRepository as dmTongHopRedisRepository;
```
Comment (thien-dang-robustaeng) [inline app/Services/DanhMucTongHopService.php]: DanhMucTongHopRedisRepository

File: app/Repositories/Redis/DanhMuc/DmTongHopRedisRepository.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    
+    
+    // get list
+    public function getList($limit, $page, $keyword, $status) {
+        $suffix = '*'; 
+        return $this->find($suffix);
+    }
+    
+    //get id
+    public function getDataDanhMucTongHopById($Khoa,$Id)
```
Comment (thien-dang-robustaeng) [inline app/Repositories/Redis/DanhMuc/DmTongHopRedisRepository.php]: No need at the moment

File: app/Repositories/Redis/DanhMuc/DmTongHopRedisRepository.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $suffix = $Khoa . ':' .$Id; 
+
+        return $this->find($suffix); 
+    }
+  
+    public function deleteDanhMucTongHop($khoa,$Id)
```
Comment (thien-dang-robustaeng) [inline app/Repositories/Redis/DanhMuc/DmTongHopRedisRepository.php]: $Id -> $id

File: app/Repositories/Redis/Hsba/HsbaKhoaPhongRepository.php [hunk 5]
```diff
@@ -17,8 +17,8 @@ public function _init(){
         parent::init(HsbaKhoaPhongRedis::class, HsbaKhoaPhongRedis::HASH_TYPE, $prefix);
     }
     
-    public function getList($phongId, $benhVienId, $startDay, $endDay, $limit = 20, $page = 1, $keyword = '', $status = -1) {
-        $suffix = '*2018-11-18';
+    public function getList( $phongId, $benhVienId, $startDay, $endDay, $limit = 20, $page = 1, $keyword = '', $status = -1) {
```
Comment (thien-dang-robustaeng) [inline app/Repositories/Redis/Hsba/HsbaKhoaPhongRepository.php]: revert

File: app/Http/Controllers/Api/V1/SamplePatientController.php [hunk 6]
```diff
@@ -82,8 +82,7 @@ public function index(Request $request)
             $messageObjects = $this->sqsRepo->pop(8);
              
             // foreach ($messageObjects as $k => $messageObject) {
-                
-                
+
```
Comment (thien-dang-robustaeng) [inline app/Http/Controllers/Api/V1/SamplePatientController.php]: revert

File: app/Repositories/Redis/BaseRedisRepository.php [hunk 7]
```diff
@@ -24,7 +24,7 @@ public function setModel($model){
         $this->model = app()->make(
            $model
         );
-        //var_dump($this->model);
+        // var_dump($this->model);die;
```
Comment (thien-dang-robustaeng) [inline app/Repositories/Redis/BaseRedisRepository.php]: revert

File: app/Services/DanhMucTongHopService.php [hunk 8]
```diff
@@ -6,15 +6,45 @@
...
+use App\Repositories\DanhMucTongHopRepository as dmTongHopRepository;
```
Comment (thien-dang-robustaeng) [inline app/Services/DanhMucTongHopService.php]: don't use this class

**Pull request merged**